pkg/osutil/osutil.go

```diff
@@ -157,3 +157,66 @@ func replaceTemplateVariables(fileSys fs.FS, srcPath string, draftConfig *config
 
 	return []byte(fileString), nil
 }
+
+// CopyDirWithTemplates - Handles Gotemplate processing and writing
+func CopyDirWithTemplates(
+	fileSys fs.FS,
+	src, dest string,
+	variableMap map[string]string,
+	templateWriter templatewriter.TemplateWriter) error {
+
+	files, err := fs.ReadDir(fileSys, src)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range files {
+
+		if f.Name() == "draft.yaml" {
```

Comment:
💡 jsut minor not for this PR but we should think of making this some form of Constant if we are using this as string comparison.

because this will add complications like DRAFT.yaml =/= draft.yaml, I don't know if we need things like this: https://pkg.go.dev/strings#EqualFold but then for the design sake folks who have wider understanding can comment more, hence jsut fyi.

Reply:
@Vidya2606 could you pull this value out into a const